torchaudio/__init__.py

```diff
@@ -168,13 +168,13 @@ def save_encinfo(filepath: str,
     # sox stores the sample rate as a float, though practically sample rates are almost always integers
     # convert integers to floats
     if signalinfo:
-        if not isinstance(signalinfo.rate, float):
+        if signalinfo.rate and not isinstance(signalinfo.rate, float):
```

Comment:
actually this is due to [SignalInfo](https://github.com/pytorch/audio/blob/master/torchaudio/_soundfile_backend.py#L16) is defined as `None` in its variables. 
So this will check if the variable exists if so then will check if the variable `isinstance` `float`